src/crimson/osd/ops_executer.cc

```diff
@@ -383,24 +383,29 @@ OpsExecuter::execute_osd_op(OSDOp& osd_op)
       return backend.getxattr(os, osd_op);
     });
   case CEPH_OSD_OP_CREATE:
+    osd_op_params.user_modify = true;
```

Comment:
How about setting `user_modify` inside `do_write_op()`? The user-invisible modifications (like `CEPH_OSD_OP_WATCH`) could be switched to a freshly introduced `do_user_invisible_write_op()` or anything with a better name :-).